### ISSUE TITLE
`reportImplicitOverride` reports `__init__` needs `@override` for class that inherits from nothing 

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#4823]
Given the following with `reportImplicitOverride` set to `"error"`:

```python
class Thing:
    def __init__(self) -> None:
        self.bar = 1
```

Line 2 is reported as an error: `Method "__init__" is not marked as override but is overriding a method in class "object"`. I understand that this is technically true (since all classes inherit from `object` and `object` has an `__init__`), but it definitely feels strange to need `@override` for every `__init__` method.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.300

**Comment by erictraut** [comment]
Reopening in light of @stroxler's comments in the python/typing discussion.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.302, which I just published. It will also be included in a future release of pylance.
